### PR TITLE
App 3147 main page

### PR DESCRIPTION
**Pull request by joanofdart**
State issue number in your PR to close it when PR merged: issue #XXX
e.g. : "Resolve the bundle size ❤️: issue #1234"

### CODE REVIEW

File: libs/movie/src/lib/movie/form/main/content-type/content-type.component.html [hunk 1]
```diff
@@ -41,9 +19,37 @@
     </mat-form-field>
   </div>
 
-  <mat-form-field appearance="outline" class="last">
-    <mat-label>Reference</mat-label>
-    <input test-id="reference" matInput type="text" placeholder="Reference" formControlName="internalRef" />
-    <mat-hint>This reference is for you only, to help you find your content more easily.</mat-hint>
-  </mat-form-field>
+  <div fxLayout="row" fxLayoutAlign="space-between center" fxLayoutGap="24px" fxLayoutAlign="start start">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/main/content-type/content-type.component.html:22]: Use sementics tag. In this case I would say "article"

File: libs/movie/src/lib/movie/pages/main/main.component.html [hunk 2]
```diff
@@ -1,50 +1,121 @@
...
-    <mat-card>
-      <mat-card-header>
-        <mat-card-title>Content Type</mat-card-title>
-      </mat-card-header>
-      <mat-card-content>
-        <movie-form-content-type [form]="main"></movie-form-content-type>
-      </mat-card-content>
-    </mat-card>
+    <form [formGroup]="main" fxLayout="column" fxLayoutGap="24px">
+      <div class="form-reference">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/main/main.component.html:6]: You're missing "Reference" title

File: libs/movie/src/lib/movie/pages/main/main.component.html [hunk 3]
```diff
@@ -1,50 +1,121 @@
...
+          <mat-form-field appearance="outline" class="full-width">
+            <mat-label>Reference</mat-label>
+            <input test-id="reference" matInput type="text" placeholder="Reference" [formControl]="main.get('internalRef')" />
+            <mat-hint>This reference is for you only, to help you find your content more easily.</mat-hint>
+          </mat-form-field>
+        </div>
+      </div>
+
+      <div class="form-images">
+        <h1>Images</h1>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/main/main.component.html:43]: It's h3

File: libs/movie/src/lib/movie/pages/main/main.component.html [hunk 4]
```diff
@@ -1,50 +1,121 @@
...
+        <div class="content-holder">
+          <h5>Poster Image</h5>
+          <span>The poster image will be used as the main image to represent your film on the marketplace, and, more generally, for all film components requiring vertical formats on the marketplace.</span>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/main/main.component.html:46]: use "p" if it's a paragraph

File: libs/movie/src/lib/movie/pages/main/main.component.html [hunk 5]
```diff
@@ -1,50 +1,121 @@
...
+          </mat-form-field>
+          <mat-form-field appearance="outline" class="full-width">
+            <mat-label>Reference</mat-label>
+            <input test-id="reference" matInput type="text" placeholder="Reference" [formControl]="main.get('internalRef')" />
+            <mat-hint>This reference is for you only, to help you find your content more easily.</mat-hint>
+          </mat-form-field>
+        </div>
+      </div>
+
+      <div class="form-images">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/main/main.component.html:42]: I don't think you need this tag here. Remove it please. 
In css you can do : 
```scss
section+h3 {
  // Style every h3 after a section
}
```

File: libs/movie/src/lib/movie/pages/main/main.component.html [hunk 6]
```diff
@@ -1,50 +1,121 @@
...
+            [storagePath]="movieId | ref: 'movie' : 'banner'"
+            ratio="banner"
+            [form]="main.banner.get('media').get('original')">
+            <h3 type>Banner Image*</h3>
+            <p class="mat-body-2" dimensions>Ratio : 16:9</p>
+            <p class="mat-caption" format>Accepted Formats: .jpg, .png</p>
+          </drop-cropper>
+        </div>
+
+        <div class="content-holder">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/main/main.component.html:57]: use article

File: libs/movie/src/lib/movie/pages/main/main.component.html [hunk 7]
```diff
@@ -1,50 +1,121 @@
...
+          <h5>Film Page Banner Image</h5>
+          <span>The poster image will be used as a cover picture on your project page on the marketplace, and, more generally, for all film components requiring horizontal formats on the marketplace.</span>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/main/main.component.html:59]: use p

File: libs/movie/src/lib/movie/pages/main/main.component.html [hunk 8]
```diff
@@ -1,50 +1,121 @@
...
+        <h1>General Information</h1>
+        <div fxLayout="row" fxLayoutAlign="start start" fxLayoutGap="24px">
+          <mat-form-field appearance="outline" class="full-width">
+            <mat-label>Production Year</mat-label>
+            <input matInput [formControl]="main.get('releaseYear')" required type="number"/>
+            <mat-error>Please enter a 4 digit date (e.g. 2020)</mat-error>
+          </mat-form-field>
+
+          <static-select scope="MOVIE_STATUS" [control]="main.get('status')" label="Estimated/Confirmed" class="full-width">
+            <ng-template let-status>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/main/main.component.html:81]: I think this is the default behavior, but I'm not sure. What happens if you remove the ng-template ?

File: libs/movie/src/lib/movie/pages/main/main.component.scss [hunk 9]
```diff
@@ -4,6 +4,16 @@ mat-divider + * {
...
 
-mat-card {
+.full-width {
   width: 100%;
-}
\ No newline at end of file
+}
+
+form {
+  .content-holder {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/main/main.component.scss:12]: apply on section or article instead.

File: libs/movie/src/lib/movie/pages/main/main.component.scss [hunk 10]
```diff
@@ -4,6 +4,16 @@ mat-divider + * {
   padding-top: 16px;
 }
 
-mat-card {
+.full-width {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/main/main.component.scss:7]: apply on section or article instead. But maybe you won't need it even.

File: libs/movie/src/lib/movie/pages/main/main.component.html (hunk 8 above)
Comment (joanofdart) [reply libs/movie/src/lib/movie/pages/main/main.component.html]: Its part of how the component is done and the required fields.
It displays the options from the select.

File: libs/movie/src/lib/movie/pages/main/main.component.scss (hunk 10 above)
Comment (joanofdart) [reply libs/movie/src/lib/movie/pages/main/main.component.scss:7]: Need them for the form-fields as their default behavior its auto.
Also need it on static-select.

**Comment by joanofdart** [comment]
Marked for deletion
**- Movie form content type.
- Movie form total runtime.
- Movie form directors.**

File: libs/movie/src/lib/movie/form/main/content-type/content-type.component.html [hunk 11]
```diff
@@ -41,9 +19,37 @@
...
+    <mat-form-field appearance="outline">
+      <mat-label>Reference</mat-label>
+      <input test-id="reference" matInput type="text" placeholder="Reference" formControlName="internalRef" />
+      <mat-hint>This reference is for you only, to help you find your content more easily.</mat-hint>
+    </mat-form-field>
+  </div>
+
+
+
+  <!-- PRODUCTION STATUS -->
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/main/content-type/content-type.component.html:42]: Please remove if not needed.

File: libs/movie/src/lib/movie/pages/main/main.component.html [hunk 12]
```diff
@@ -1,71 +1,136 @@
...
-        <mat-card-title i18n>Festival{ festivalPrizes.get('prizes').controls.length, plural,=1 {} other {s}}</mat-card-title>
-      </mat-card-header>
-      <mat-card-content>
-        <movie-form-festival-prizes [form]="festivalPrizes"></movie-form-festival-prizes>
-      </mat-card-content>
-    </mat-card>
+    <form [formGroup]="main" fxLayout="column" fxLayoutGap="24px">
 
+      <h3>Reference</h3>
+      <section fxLayout="row" fxLayoutAlign="space-between center" fxLayoutGap="24px">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/main/main.component.html:8]: Following [this article](https://css-tricks.com/how-to-section-your-html/#:~:text=The%20heading%20always%20comes%20first,content%20inside%20that%20sectioning%20element) the right structure is :
```html
<section>
  <h3></h3>
  <article fxLayout fxLayoutGap="24px">
    <mat-form-field appearance="outline" fxFlex><mat-form-field>
    <mat-form-field appearance="outline" fxFlex><mat-form-field>
  </article>
  <article fxLayout fxLayoutGap="24px">
    <mat-form-field appearance="outline" fxFlex><mat-form-field>
    <mat-form-field appearance="outline" fxFlex><mat-form-field>
  </article>
</section>
```

File: libs/movie/src/lib/movie/pages/main/main.component.html [hunk 13]
```diff
@@ -1,71 +1,136 @@
...
+          </mat-select>
+        </mat-form-field>
+        <mat-form-field appearance="outline" class="full-width">
+          <mat-label>Reference</mat-label>
+          <input test-id="reference" matInput type="text" placeholder="Reference" [formControl]="main.get('internalRef')" />
+          <mat-hint>This reference is for you only, to help you find your content more easily.</mat-hint>
+        </mat-form-field>
+      </section>
+
+      <h3>Images</h3>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/main/main.component.html:42]: Here you just have to wrap again with section

File: libs/movie/src/lib/movie/pages/main/main.component.html [hunk 14]
```diff
@@ -1,71 +1,136 @@
...
+            ratio="poster"
+            [form]="main.poster.get('media').get('original')">
+            <h3 type>Poster*</h3>
+            <p class="mat-body-2" dimensions>Dimensions : 120 x 160 px</p>
+            <p class="mat-caption" format>Accepted Formats: .jpg, .png</p>
+          </drop-cropper>
+      </article>
+
+      <h3>General Information</h3>
+      <section fxLayout="row" fxLayoutAlign="start start" fxLayoutGap="24px">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/main/main.component.html:70]: Here it'll look like that : 
```html
<section>
  <h3></h3>
  <article fxLayout fxLayoutGap="24px">
    <mat-form-field appearance="outline" fxFlex></>
    <static-select fxFlex ....></>
  </article>
  <article>
    <chips-autocomplete ...></>
    ....
  </article>
  <article fxLayout fxLayoutGap="24px">
    <static-select fxFlex ...></>
    <mat-form-field  fxFlex appearance="outline">
  </article>
</section>
```

File: libs/movie/src/lib/movie/pages/main/main.component.html [hunk 15]
```diff
@@ -1,71 +1,136 @@
...
+          <input matInput type="number" min="0" [formControl]="main.get('totalRunTime')" name="runningTime"/>
+          <mat-hint>In minutes</mat-hint>
+          <mat-error *ngIf="main.hasError('min')">
+            Only positive values are accepted.
+          </mat-error>
+        </mat-form-field>
+      </section>
+
+
+      <h3>Directors</h3>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/pages/main/main.component.html:108]: Here you just need to wrap inside a section.

File: libs/movie/src/lib/movie/pages/main/main.component.html (hunk 8 above)
Comment (GrandSchtroumpf) [reply libs/movie/src/lib/movie/pages/main/main.component.html:84]: So you could just do : 
```html
<static-select scope="RELEASE_STATUS" [control]="main.get('status')" label="Estimated/Confirmed"></static-select>
```
Without the `ng-template` ? I'm sorry I'm not sure I understood your message above.

**Comment by joanofdart** [comment]
Marked for deletion
**- Movie form content type.**
**- Movie form total runtime.**
**- Movie form directors.**
**- Form display name**

**Pull request merged**